Sources/ArgumentParser/Completions/ZshCompletionsGenerator.swift

```diff
@@ -12,12 +12,13 @@
 struct ZshCompletionsGenerator {
   /// Generates a Zsh completion script for the given command.
   static func generateCompletionScript(_ type: ParsableCommand.Type) -> String {
-    let initialFunctionName = [type].completionFunctionName()
+    let initialFunctionName = [type].completionFunctionName().zshEscapingCommandName()
+    let commandName = type._commandName.zshEscapingCommandName()
 
     return """
-    #compdef \(type._commandName)
+    #compdef \(commandName)
     local context state state_descr line
-    _\(type._commandName)_commandname=$words[1]
+    _\(commandName)_commandname=$words[1]
```

Comment:
As far as I understand it, this is the only part that needs to change, since dashes _are_ allowed in the function names, just not as a variable name. Can you still reproduce the issue if you only change this part here? As an extra benefit, this smaller change wouldn't cause problems for people who declare both `sub-command` and `sub_command` subcommands (which is a bad idea, but it will surely happen eventually).

Reply:
You are right,  this is the only required changes. I made the change and tried locally :) Thanks for catching it @natecook1000. PR updated!